pkg/graveler/graveler.go

```diff
@@ -1655,10 +1691,12 @@ func (g *Graveler) Delete(ctx context.Context, repository *RepositoryRecord, bra
 	}
 
 	log := g.log(ctx).WithFields(logging.Fields{"key": key, "operation": "delete"})
-	return g.safeBranchWrite(ctx, log, repository, branchID,
+	retries, err := g.safeBranchWrite(ctx, log, repository, branchID,
 		safeBranchWriteOptions{}, func(branch *Branch) error {
 			return g.deleteUnsafe(ctx, repository, branch, key, nil)
 		})
+	g.monitorRetries(ctx, retries, repository.RepositoryID, branchID, "delete")
```

Comment:
I do feel like this pattern should become the function itself or at least we should wrap this common functionality with another func:
```suggestion
	err := g.safeBranchWriteWithMonitor(ctx, log, repository, branchID,
		safeBranchWriteOptions{}, func(branch *Branch) error {
			return g.deleteUnsafe(ctx, repository, branch, key, nil)
		}
```

or just 
```suggestion
          err := g.safeBranchWrite(ctx, log, repository, branchID,
		safeBranchWriteOptions{}, func(branch *Branch) error {
			return g.deleteUnsafe(ctx, repository, branch, key, nil)
		})
```

Reply:
Mostly worried about adding `safeBranchWrite` in the future without monitoring the retries.